Fix issue 29: correct implementation of evalBv

maurobringolf opened the pull request:
Fixes #29, e.g. `evalBv`. As far as I can tell, the current implementation proceeds as follows:

1. Evaluate the given AST (`eval`)
2. Apply the conversion function from bit vector to Integer (`mkBv2int`)
3. Try to extract the Integer from the AST (`getInt`)

As a result, the conversion is never evaluated and the integer extraction fails because the AST node is a function application and not an integer. This PR swaps order of operations and adds the conversion function to the AST before evaluating it.

At least the example given in the issue works fine after this change.

IagoAbal reviewed (approved):
Thanks for the patch! I trust this works. But then maybe we should drop `getBv`?

maurobringolf commented:
Agreed, I did not realize that this was its only usage and removed it now.

In #44 I added a new testsuite for regression tests which I used to test this PR against the original issue script. If you are fine with this test setup I can include the regression test in this PR as well.

IagoAbal commented:
> Agreed, I did not realize that this was its only usage and removed it now.
> 
> In #44 I added a new testsuite for regression tests which I used to test this PR against the original issue script. If you are fine with this test setup I can include the regression test in this PR as well.

Yes please, that would be great!

maurobringolf commented:
Added a simplified version of the regression test and also a quickcheck test for checking that `evalBv` behaves as the inverse of `mkBitvector`, just like the existing tests.

IagoAbal commented:
Could you rebase your branch against master? It now conflicts with the changes done to PR #44.

maurobringolf commented:
Just finished rebasing :+1:
